### PR TITLE
Support Phabricator stack of patches 

### LINKED ISSUE #1215: shipit_static_analysis: Support a stack of revisions
We should support stack of revisions too (when the developer manually sets dependencies between them).

### PR DESCRIPTION
**Pull request by La0**
Fixes #1215

### CODE REVIEW

File: lib/cli_common/cli_common/phabricator.py [hunk 1]
```diff
@@ -163,6 +163,28 @@ def comment(self, revision_id, message):
             attach_inlines=1,
         )
 
+    def load_parents(self, revision_phid):
```
Comment (marco-c) [inline lib/cli_common/cli_common/phabricator.py]: There's an implementation that might be faster (smaller amount of queries) that smacleod gave me a while ago: https://github.com/mozilla/code-coverage-addon/issues/6#issuecomment-382797648.
If you want to do it in a follow-up, can you file an issue?

File: src/staticanalysis/bot/static_analysis_bot/revisions.py [hunk 2]
```diff
@@ -134,33 +134,74 @@ def url(self):
...
-        assert isinstance(repo, hglib.client.hgclient)
+        parents = self.api.load_parents(self.phid)
+        if parents:
+
+            # Load all parent diffs
+            for parent in parents:
+                logger.info('Loading parent diff', phid=parent)
+                parent_diffs = self.api.search_diffs(revision_phid=parent)
+
+                # Use only the first (last in time) diff for each revision
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/revisions.py]: Two questions:
1) Maybe we should get the parent of the specific diff? There could be cases where the analysis is triggered, then the developer uploads a new stack of revisions and we get wrong results. I'm sure this is really rare, but in theory it could happen.
2) If we don't do (1), we should file an issue about it so we don't forget. Then, are we sure the first is always the last in time? Maybe we should sort by ID so that we're sure we're getting the latest?

**Comment by marco-c** [comment]
We should also add some tests for this, could you either do it in this PR or file a follow-up so we don't forget?

File: lib/cli_common/cli_common/phabricator.py (hunk 1 above)
Comment (La0) [reply lib/cli_common/cli_common/phabricator.py]: This code sample is only to get the first parents and their details, but will not reach older parents than depth=1.

File: src/staticanalysis/bot/static_analysis_bot/revisions.py (hunk 2 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/revisions.py]: 1 is not possible, `edge.search` only gives the revisions
I'm doing 2

**Comment by La0** [comment]
I created the issue #1463 to add some unit tests + fixtures on `edge.search`.
